lib/core/src/persist/mod.rs

```diff
@@ -375,6 +375,21 @@ impl Persister {
             .optional()?)
     }
 
+    pub fn get_payment_by_destination(&self, destination: &str) -> Result<Option<Payment>> {
+        Ok(self
+            .get_connection()?
+            .query_row(
+                &self.select_payment_query(
+                    Some("(rs.invoice = ?1 OR ss.invoice = ?1 OR cs.claim_address = ?1 OR pd.destination = ?1)"), 
```

Comment:
A few questions:
- If the getter says `by_destination`, shouldn't it ideally just search in `pd.destination`? AFAIK `destination` is documented as "invoice or Liquid BIP21/address", so it should cover the invoice as well.
- If it's a generic getter `by_destination` and someone used a BIP21 as destination, but now queries by the onchain address from that BIP21, this won't find it, correct? If so, I'm wondering whether it should find it, since technically the onchain address *is* the destination.

What do you think? cc @hydra-yse

Reply:
I tried with the `pd.destination` but a payment_details table entry is only added for payments to a Liquid address. The columns used to otherwise find the `destination` are the columns otherwise used to fill the PaymentDetails for the Payment.